Realm/RLMSyncPermissionValue.mm

```diff
@@ -0,0 +1,201 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMSyncPermissionValue_Private.hpp"
+
+#import "RLMUtil.hpp"
+
+using namespace realm;
+using ConditionType = Permission::Condition::Type;
+
+namespace {
+
+Permission::AccessLevel accessLevelForObjcAccessLevel(RLMSyncAccessLevel level) {
+    switch (level) {
+        case RLMSyncAccessLevelNone:
+            return Permission::AccessLevel::None;
+        case RLMSyncAccessLevelRead:
+            return Permission::AccessLevel::Read;
+        case RLMSyncAccessLevelWrite:
+            return Permission::AccessLevel::Write;
+        case RLMSyncAccessLevelAdmin:
+            return Permission::AccessLevel::Admin;
+    }
+    REALM_UNREACHABLE();
+}
+
+RLMSyncAccessLevel objCAccessLevelForAccessLevel(Permission::AccessLevel level) {
+    switch (level) {
+        case Permission::AccessLevel::None:
+            return RLMSyncAccessLevelNone;
+        case Permission::AccessLevel::Read:
+            return RLMSyncAccessLevelRead;
+        case Permission::AccessLevel::Write:
+            return RLMSyncAccessLevelWrite;
+        case Permission::AccessLevel::Admin:
+            return RLMSyncAccessLevelAdmin;
+    }
+    REALM_UNREACHABLE();
+}
+
+NSString *descriptionForAccessLevel(RLMSyncAccessLevel level) {
```

Comment:
I would say yes for the first two, no for the second two (since Cocoa needs to compare both "output" permissions (those backed by the object store) and "input" permissions (those whose state is set using ivars)).